src/stores/tournament.store.ts

```diff
@@ -13,79 +13,200 @@ const groupBy = <T extends object, K extends keyof T>(items: T[], key: K) => ite
 }), {} as Record<T[K], T[]>);
 
 export const useTournamentStore = defineStore('tournament', () => {
-  const events = ref<Record<number, Event[]>>({});
-  const event = ref<Event[]>([]);
+  const events = ref<Record<number, Event>>({});
   const tournaments = ref<Record<number, Tournament>>({});
-  const tournament = ref<Tournament>();
 
-  async function fetchEvents() {
+  async function fetchAllEvents() {
     try {
       const res = await axios.get<Event[]>('/tournament/event');
-      const ev = res.data;
-      events.value = groupBy(ev, 'year');
+      res.data.forEach((ev) => { events.value[ev.id] = ev; });
       // console.log(years)
-    } catch (error) {
+    } catch (err: any) {
       // console.log(error);
     }
   }
 
+  async function fetchEventsByYear(year: number) {
+    try {
+      const res = await axios.get<Event[]>(`tournament/event/year/${year}`);
+      res.data.forEach((ev) => { events.value[ev.id] = ev; });
+    } catch (err: any) {
+      /* empty */
+    }
+  }
+
+  async function fetchEventsByYears(years: number[]) {
+    await Promise.all<void>(years.map<Promise<void>>((year) => fetchEventsByYear(year)));
+  }
+
   async function fetchThisYear() {
     const year = new Date().getFullYear();
-    const res = await axios.get<Event[]>(`/tournament/event/year/${year}`);
-    event.value = res.data;
-    // console.log(res.data)
+    await fetchEventsByYear(year);
   }
 
-  async function fetchTournaments(ids: number[]) {
-    const promises = ids.map<Promise<Tournament | Error>>((id) => axios.get<Tournament>(`/tournament/tournament/${id}`).then((res) => res.data).catch((err: Error) => err));
+  async function fetchEventById(id: number) {
+    try {
+      const res = await axios<Event>(`tournament/event/${id}`);
+      events.value[id] = res.data;
+    } catch (err: any) {
+      /* empty */
+    }
+  }
 
-    const data = await Promise.all<Tournament | Error>(promises);
+  async function fetchEventByIds(ids: number[]) {
+    await Promise.all<void>(ids.map<Promise<void>>((id) => fetchEventById(id)));
```

Comment:
Further improvement could be to add a backend endpoint doing this fetch instead of using several backend calls but this is fine atm.